uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/SearchRESTController.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Licensed to Apereo under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright ownership. Apereo
+ * licenses this file to you under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the License at the
+ * following location:
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apereo.portal.rest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.http.HttpHeaders;
+import org.apereo.portal.portlet.om.IPortletDefinition;
+import org.apereo.portal.portlet.registry.IPortletDefinitionRegistry;
+import org.apereo.portal.portlets.lookup.PersonLookupHelperImpl;
+import org.apereo.portal.rest.utils.PortletRegistryUtil;
+import org.apereo.portal.security.IPerson;
+import org.apereo.portal.security.IPersonManager;
+import org.apereo.services.persondir.IPersonAttributes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+/**
+ * REST Search endpoint that aggregates search of various types.
+ *
+ * <p>JSON results example:
+ *
+ * <p>{"people" : [...], "portlets" : [...] }
+ *
+ * @since 5.0
+ */
+@Controller
+@RequestMapping("/search")
+public final class SearchRESTController {
+
+    private static final Logger logger = LoggerFactory.getLogger(SearchRESTController.class);
```

Comment:
I think `private final Logger logger = LoggerFactory.getLogger(getClass());` is better.